api/app/src/shared/analytics.ts

```diff
@@ -0,0 +1,32 @@
+import { PostHog } from "posthog-node";
+import { Request } from "express";
+
+import { Config } from "./config";
+import { getCxId } from "../routes/util";
+
+export const queryAnalytics = ({
+  message,
+  req,
+  apiType,
+}: {
+  message: string;
+  req: Request;
+  apiType: string;
+}): void => {
+  const cxId = getCxId(req);
+
+  if (!cxId) {
+    return;
+  }
+
+  analytics.capture({
+    distinctId: cxId,
+    event: message,
+    properties: {
+      environment: Config.getEnvironment(),
+      apiType,
+    },
+  });
```

Comment:
When doing asynchronous coding, I suggest we have at least a `.catch()` block logging whatever error happens, so we have a way to see them on logs.

Reply:
OTOH, I understand if we want not to have any error handling code to analytics code, since its NABD to lose some entries if it fails to capture/process those events.